glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt

```diff
@@ -43,14 +43,18 @@ internal class UploadBody(
     }
 }
 
-@Structure.FieldOrder("tag", "upload")
 internal open class FfiPingUploadTask(
     // NOTE: We need to provide defaults here, so that JNA can create this object.
     @JvmField var tag: Byte = UploadTaskTag.Done.ordinal.toByte(),
     @JvmField var upload: UploadBody = UploadBody()
 ) : Union() {
     class ByValue : FfiPingUploadTask(), Structure.ByValue
 
+    init {
+        // Initialize to be the `tag`-only variant
+        setType("tag")
```

Comment:
Interesting. Why was this working before?

Reply:
Because that field is valid in both variants (it's also the first 8 bits in the `UploadBody`, therefore the first 8 bits of the union are always an integer).
This just makes it more explicit, which IMO helps us still understand this code in 2, 4 and 8 weeks.

Reply:
Interesting! Thanks for clarifying!